src/controllers/observation-controller.ts

```diff
@@ -36,152 +67,305 @@ export default class ObservationController extends AbstractController {
+   * @param {number} pageSize Page size, default value is set in config file
+   * @param {SortingDirection} sortingDirection Sorting direction
+   * @param {string} sortingColumn Column to search, can be any of ObservationDto field name
+   */
+  @GET
+  @Path('/')
+  @Response<ObservationsListResponse>(200, 'List of all available observations.')
+  @Response<CustomError>(401, 'Unauthorised.')
+  public async getObservations(
+    @ContextRequest req: RequestWithPageParams,
+    @QueryParam('lang') lang: string = 'eng',
+    @QueryParam('pageNumber') pageNumber: number = 0,
+    @QueryParam('pageSize') pageSize: number = 5,
+    @QueryParam('sortingDirection') sortingDirection: SortingDirection = SortingDirection.asc,
+    @QueryParam('sortingColumn') sortingColumn?: string,
+  ): Promise<ObservationsListResponse> {
+    const langOrigin = LocaleOrigin[lang] ? LocaleOrigin[lang] : 'desc_eng';
+
+    const paramsSearch = parsePageParams({ pageNumber, pageSize, sortingColumn, sortingDirection });
+    const paramsAggregation = parseWhereParams(req.query, req.user);
+    const observations = await this.observations.findAndCount(Object.assign(paramsSearch, paramsAggregation));
+
+    const content = observations[0].map(obs => {
+      // sanitaze user's data
+      const finder = Object.assign({}, obs.finder.sanitizeUser(), { id: obs.finder.id });
+      // transform 'observation'
+      const observation = Object.entries(obs)
+        // clear 'filter' field
+        .filter(([ObservationField]) => ObservationField !== 'ring')
+        // map 'lang' param according 'Locale'
+        .map(([ObservationKey, ObservationValue]) => {
+          if (typeof ObservationValue === 'object' && ObservationValue !== null) {
+            const value = Object.entries(ObservationValue)
+              .filter(([subfield]) => filterFieldByLocale(subfield as Locale, langOrigin))
+              .map(([subfield, subValue]) =>
+                subfield === langOrigin ? ['desc', subValue] : ([subfield, subValue] as any),
+              )
+              .reduce((acc, [subfield, subValue]) => Object.assign(acc, { [subfield]: subValue }), {});
+            return [ObservationKey, value];
+          }
+          return [ObservationKey, ObservationValue];
+        })
+        .reduce(
+          (acc, [ObservationField, ObservationValue]) => Object.assign(acc, { [ObservationField]: ObservationValue }),
+          {},
+        ) as ObservationDto;
+      return Object.assign({}, observation, { finder }, { ring: obs.ring && obs.ring.id });
+    });
+
+    return {
+      content,
+      pageNumber: paramsSearch.number,
+      pageSize: paramsSearch.size,
+      totalElements: observations[1],
+    };
+  }
+
+  // TODO: currently typescript-rest doesn't support [key in keyof smth] syntax
+  /**
+   * Get all available values for filtering observations.
+   * Returns all existing values of observation fields.
+   */
+  @GET
+  @Path('/aggregations')
```

Comment:
I faced  with issue, that when I request this endpoint also is called **get** `observations/:id` which throws error. I think that it's issue with `typescript-rest` itself. Usually express declarations of routes

```js
router.get('/route', handler1);
router.get('/:id', handler2);
```
will never pass down to flow route handlers execution, with one exception if `handler1` contains `next()` statement

Please investigate this issue and fix it

Reply:
https://github.com/thiagobustamante/typescript-rest/issues/78

Reply:
Fixed